libcrux-ml-kem/src/ind_cca.rs

```diff
@@ -261,16 +262,15 @@ pub(crate) fn decapsulate<
         Hasher,
     >(ind_cpa_public_key, decrypted, pseudorandomness);
 
-    let selector = compare_ciphertexts_in_constant_time(ciphertext.as_ref(), &expected_ciphertext);
-
     let implicit_rejection_shared_secret =
         Scheme::kdf::<K, CIPHERTEXT_SIZE, Hasher>(&implicit_rejection_shared_secret, ciphertext);
     let shared_secret = Scheme::kdf::<K, CIPHERTEXT_SIZE, Hasher>(shared_secret, ciphertext);
 
-    select_shared_secret_in_constant_time(
+    compare_ciphertexts_select_shared_secret_in_constant_time(
```

Comment:
I think this should be fine. But I wonder why F* has trouble here. @karthikbhargavan do you understand what's going on here?